ldelf/ta_elf.c

```diff
@@ -1139,6 +1156,9 @@ static void load_main(struct ta_elf *elf)
 	save_symtab(elf);
 	close_handle(elf);
 	set_tls_offset(elf);
+	parse_property_segment(elf);
+	if (elf->bti_enabled)
+		ta_elf_add_bti(elf);
```

Comment:
I assume that the reason we're doing this so late is that we can't parse the property segment earlier.

Reply:
Yes, the property segment can be parsed only after the load segments have been mapped.